api/src/pcapi/core/users/api.py

```diff
@@ -1790,3 +1791,46 @@ def enable_new_pro_nav(user: models.User) -> None:
     pro_new_nav_state.newNavDate = datetime.datetime.utcnow()
     db.session.add(pro_new_nav_state)
     db.session.commit()
+
+
+def clean_gdpr_extracts() -> None:
+    extension = "zip"
+    folder = settings.GCP_GDPR_EXTRACT_FOLDER
+    files = object_storage.list_files(
+        folder=folder,
+        bucket=settings.GCP_GDPR_EXTRACT_BUCKET,
+    )
+    files_ids = set()
+    for file_path in files:
+        try:
+            file_name = file_path.split("/")[-1]
+            extract_id = int(file_name[: -(len(extension) + 1)])
+            files_ids.add(extract_id)
+        except ValueError:
+            continue
+
+    ids_in_db_query = models.GdprUserDataExtract.query.filter(
+        models.GdprUserDataExtract.id.in_(files_ids)
+    ).with_entities(models.GdprUserDataExtract.id)
+    ids_in_db = {r.id for r in ids_in_db_query}
+
+    if files_ids != ids_in_db:
```

Comment:
A priori cette condition n'est pas utile car ce sera bien géré dans la différence dans le `for`.